anyscale_provider/operators/anyscale.py

```diff
@@ -389,40 +386,38 @@ def execute(self, context: Context) -> None:
-
-    def execute_complete(self, context: Context, event: Any) -> None:
-        """
-        Complete the execution of the service rollout based on the trigger event.
+        # Failure states based on https://docs.anyscale.com/reference/service-api#servicestate
+        # Keeping in mind that TERMINATED is expected, temporarily, since this is a rollout operation.
+        # Once the service is being rolled out, it is also expected it may be temporarily UNHEALTHY.
+        failure_states = [ServiceState.SYSTEM_FAILURE, ServiceState.UNKNOWN]
 
-        This method is called when the trigger fires and provides the final status
-        of the service rollout. It raises an exception if the rollout failed.
+        # TODO: Leverage the Airflow triggerer mechanism to poll the job status.
+        # This can only happen once we can check asynchronously the Anyscale job status,
+        # something that is currently not possible with the latest Anyscale SDK version (0.26.75).
+        # A possible path forward is to use the Anyscale API directly to enable this capability.
 
-        :param context: The Airflow context.
-        :param event: The event data from the trigger.
-        :return: None
-        """
-        service_name = event["service_name"]
-        state = event["state"]
-
-        self.log.info(f"Execution completed for service {service_name} with state: {state}")
-
-        if state == ServiceState.SYSTEM_FAILURE:
-            error_message = event.get("message", "")
-            error_msg = f"Anyscale service deployment {service_name} failed with error: {error_message}"
-            self.log.error(error_msg)
-            raise AirflowException(error_msg)
+        if self.wait_for_completion:
+            has_succeeded = False
+            for _ in range(int(self.service_rollout_timeout_seconds // self.poll_interval)):
+                service_status = self.hook.get_service_status(name=self.name)
+                current_state = str(service_status.state)
+                self.log.info(f"Current service state for {self.name} is: {current_state}")
+                if current_state == ServiceState.RUNNING:
+                    self.log.info(f"Service {self.name} is healthy and running.")
+                    has_succeeded = True
+                    break
+                elif current_state in failure_states:
+                    raise AirflowException(f"Service {self.name} failed.")
+                time.sleep(self.poll_interval)
+            if not has_succeeded:
+                raise AirflowException(
+                    f"Service {self.name} timed out after {self.service_rollout_timeout_seconds} seconds. The service was not complete within the desired service_rollout_timeout_seconds: {self.service_rollout_timeout_seconds}."
+                )
         else:
-            self.log.info(f"Anyscale service deployment {service_name} completed successfully")
+            service_status = self.hook.get_service_status(name=self.name)
```

Comment:
```suggestion
            service_status = self.hook.get_service_status(service_name=self.name)
```
if we can keep the hook arg as was before